dipy/align/tests/test_transforms.py

```diff
@@ -0,0 +1,230 @@
+    expected = np.array([[1, 0, 0, dx],
+                         [0, 1, 0, dy],
+                         [0, 0, 1, dz],
+                         [0, 0, 0, 1]])
+    actual = transform.param_to_matrix(theta)
+    assert_array_equal(actual, expected)
+
+    # Test rotation matrix 3D
+    transform = regtransforms[('ROTATION', 3)]
+    theta = np.random.rand(3)
+    ca = np.cos(theta[0])
+    sa = np.sin(theta[0])
+    cb = np.cos(theta[1])
+    sb = np.sin(theta[1])
+    cc = np.cos(theta[2])
+    sc = np.sin(theta[2])
+
+    X = np.array([[1,  0,  0 ],
+                  [0, ca, -sa],
+                  [0, sa,  ca]])
+    Y = np.array([[cb,  0, sb],
+                  [0,   1,  0],
+                  [-sb, 0, cb]])
+    Z = np.array([[cc, -sc, 0],
+                  [sc,  cc, 0],
+                  [0 ,   0, 1]])
+
+    R = Z.dot(X.dot(Y)) # Apply in order: Y, X, Z (Y goes to the right)
+    expected = np.eye(4)
+    expected[:3, :3] = R[:3, :3]
+    actual = transform.param_to_matrix(theta)
+    assert_array_almost_equal(actual, expected)
+
+    # Test rigid matrix 3D
+    transform = regtransforms[('RIGID', 3)]
+    theta = np.random.rand(6)
+    ca = np.cos(theta[0])
+    sa = np.sin(theta[0])
+    cb = np.cos(theta[1])
+    sb = np.sin(theta[1])
+    cc = np.cos(theta[2])
+    sc = np.sin(theta[2])
+
+    X = np.array([[1,  0,  0 ],
+                  [0, ca, -sa],
+                  [0, sa,  ca]])
+    Y = np.array([[cb,  0, sb],
+                  [0,   1,  0],
+                  [-sb, 0, cb]])
+    Z = np.array([[cc, -sc, 0],
+                  [sc,  cc, 0],
+                  [0 ,   0, 1]])
+
+    R = Z.dot(X.dot(Y)) # Apply in order: Y, X, Z (Y goes to the right)
+    expected = np.eye(4)
+    expected[:3, :3] = R[:3, :3]
+    expected[:3, 3] = theta[3:6]
+    actual = transform.param_to_matrix(theta)
+    assert_array_almost_equal(actual, expected)
+
+    # Test scaling matrix 2D
+    transform = regtransforms[('SCALING', 3)]
+    factor = np.random.rand()
+    theta = np.array([factor])
+    expected = np.array([[factor, 0, 0, 0],
+                         [0, factor, 0, 0],
+                         [0, 0, factor, 0],
+                         [0, 0, 0, 1]])
+    actual = transform.param_to_matrix(theta)
+    assert_array_almost_equal(actual, expected)
+
+    # Test affine 2D
+    transform = regtransforms[('AFFINE', 3)]
+    theta = np.random.rand(12)
+    expected = np.eye(4)
+    expected[0,:] = theta[:4]
+    expected[1,:] = theta[4:8]
+    expected[2,:] = theta[8:12]
+    actual = transform.param_to_matrix(theta)
+    assert_array_almost_equal(actual, expected)
+
+    # Verify that ValueError is raised if incorrect number of parameters
+    for transform in regtransforms.values():
+        n = transform.get_number_of_parameters()
+
+        # Set incorrect number of parameters
+        theta = np.zeros(n + 1, dtype=np.float64)
+        assert_raises(ValueError, transform.param_to_matrix, theta)
+
```

Comment:
Just a remark here. Have in mind that we have the compose_matrix and decompose_matrix functions in 
dipy.core.geometry which enable to check if your matrix composition is correct.
https://github.com/nipy/dipy/blob/master/dipy/core/geometry.py

However, the parametrization is different for the affine case than yours. We have
the shears and rotations independently. It would be interesting to have a function 
that translates your parametrization to the parametrization used in core.geometry
and SLR accordingly. 
This is only a suggestion not a required remark. But, I was thinking that it could help
us double check the numerical stability of your transforms. I played quite a bit
with the de/compose_matrix functions and I found them very stable numerically.
I think Matthew put quite some thinking and testing in those functions.

Anyway, have it in the back of your mind.